website/docs/r/machine.html.markdown

```diff
@@ -9,3 +9,135 @@ description: |-
+  region           = var.region
+}
+
+data "vra_zone" "this" {
+  name = var.zone
+}
+
+data "vra_project" "this" {
+  name = var.project
+}
+
+resource "vra_flavor_profile" "this" {
+  name        = "tf-vra-flavor-profile"
+  description = "my flavor"
+  region_id   = data.vra_region.this.id
+
+  flavor_mapping {
+    name          = "small"
+    instance_type = "t2.small"
+  }
+
+  flavor_mapping {
+    name          = "medium"
+    instance_type = "t2.medium"
+  }
+}
+
+resource "vra_image_profile" "this" {
+  name        = "tf-vra-image-profile"
+  description = "terraform test image profile"
+  region_id   = data.vra_region.this.id
+
+  image_mapping {
+    name       = "ubuntu"
+    image_name = var.image_name
+  }
+}
+
+resource "vra_machine" "this" {
+  name        = "tf-machine"
+  description = "terrafrom test machine"
+  project_id  = data.vra_project.this.id
+  image       = "ubuntu"
+  flavor      = "small"
+
+  tags {
+    key   = "foo"
+    value = "bar"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported for a machine resource:
+
+* `flavor` - (Required) Flavor of machine instance
+
+* `image` - (Required) The type of image used for this machine.
+
+* `name` - (Required) A human-friendly name used as an identifier in APIs that support this option.
+
+* `project_id` - (Required) The id of the project the current user belongs to.
+
+* `boot_config` - (Optional) Machine boot config that will be passed to the instance that can be used to perform common automated configuration tasks and even run scripts after the instance starts. It is nested argument with the following property.
+  * `content` - A valid cloud config data in json-escaped yaml syntax
+
+* `constraints` - (Optional) Constraints that are used to drive placement policies for the virtual machine that is produced from this specification. Constraint expressions are matched against tags on existing placement targets.
+example:[{"mandatory" : "true", "expression": "environment":"prod"}, {"mandatory" : "false", "expression": "pci"}]. It is nested argument with the following properties.
+  * `expression` - A constraint that is conveyed to the policy engine. An expression of the form "[!]tag-key[:[tag-value]]", used to indicate a constraint match on keys and values of tags. 
+  * `mandatory` - Indicates whether this constraint should be strictly enforced or not.
+
+* `custom_properties` - (Optional) Additional custom properties that may be used to extend the machine.
+
+* `description` - (Optional) Describes machine within the scope of your organization and is not propagated to the cloud.
+
+* `disks` - (Optional) A set of disk specifications for this machine. It is nested argument with the following properties.
+  * `block_device_id` - The id of the existing block device.
+  * `description` - A human-friendly description.
+  * `name` - A human-friendly name used as an identifier in APIs that support this option.
```

Comment:
Please specify which of these properties are optional vs. required, if disks is provided in the config. Same for other nested properties.

Reply:
Done